docs/guides/react-compiler.mdx

```diff
@@ -0,0 +1,31 @@
+---
+slug: guides/react-compiler
+title: Enabling React Compiler in Waku
+description: Learn how to enable the React Compiler in your Waku project using the Vite Plugin to optimize your React applications.
+---
+
+The React Compiler is a build-time tool designed to optimize React applications automatically. This guide will show you how to enable the React Compiler by integrating it with the `@vitejs/plugin-react`.
+
+```ts
+// vite.config.ts
+import react from '@vitejs/plugin-react';
+
+const ReactCompilerConfig = { /* ... */ };
+
+export default defineConfig(() => {
+  return {
+    plugins: [
+      react({
+        babel: {
+          plugins: [
+            ["babel-plugin-react-compiler", ReactCompilerConfig],
+          ],
+        },
+      }),
+    ],
+    // ...
+  };
+});
+```
+
+For a practical demonstration, visit this [StackBlitz demo](https://stackblitz.com/edit/github-ukt9tm?file=vite.config.ts) which provides a real-world example of the configuration in action.
```

Comment:
I've updated the demo.

```suggestion
For a practical demonstration, visit this [StackBlitz demo](https://stackblitz.com/edit/github-h4uxwo?file=vite.config.ts) which provides a real-world example of the configuration in action.
```

As I tried, removing "annotation" will cause errors. Can you please try it? I guess we should add it in the instruction above. What do you think? Do you think we can recommend it even with "annotation"?

Reply:
![image](https://github.com/user-attachments/assets/57fe2a1d-cf53-4912-89a8-77261d746a5c)

I believe we shouldn't add this restriction because if we add an "annotation", users will need add the "use memo" directive manually

I found the error `Error: __vite_ssr_import_1__.c is not a function`, actually I encountered this error when I implement react compiler feature, I can try to fix it issue

Reply:
> I can try to fix it issue

That will be great.

Reply:
![image](https://github.com/user-attachments/assets/a83b17fd-aafe-47bd-9cca-b85d1afd15b6)

This issue occurs when I pass react compiler config in. So in my [implementation](https://github.com/dai-shi/waku/pull/853) I only pass react compiler config in `viteReact` called by `createMainViteServer`

Reply:
Do we not need react compiler for server files?

My plan is to allow customizing vite config for each server in v0.22.0. Give me more time.

Reply:
Actually, I noticed the change isn't breaking much, so I decided to work on it during v0.21.*. 

https://github.com/dai-shi/waku/issues/24#issuecomment-2564364866  @rmarscher